### PR TITLE
[WIP] Upgrade protractor and selenium to the latest versions

### PR DESCRIPTION
**Pull request by jwforres**

### CODE REVIEW

**Comment by jwforres** [comment]
@benjaminapetersen @spadgett FYI

File: test/integration/helpers.js [hunk 1]
```diff
@@ -22,19 +23,16 @@ exports.login = function(loginPageAlreadyLoaded) {
   // The login page doesn't use angular, so we have to use the underlying WebDriver instance
   var driver = browser.driver;
   if (!loginPageAlreadyLoaded) {
-    browser.get('/');
+    driver.get('/');
```
Comment (jwforres) [inline test/integration/helpers.js]: So this is where things currently seem to be breaking, this get never resolves.

Comment (benjaminapetersen) [inline test/integration/helpers.js]: yeah this is weird. The [webdriver docs](http://webdriver.io/api/protocol/url.html) seem ton indicate we should still have `browser.url`/`driver.url`...

But when I `Object.keys(driver)` I see:
```
[ 'flow_',
  'session_',
  'executor_',
  'fileDetector_',
  'onQuit_',
  'cancel',
  'then',
  'catch',
  'getNetworkConnection',
  'setNetworkConnection',
  'toggleAirplaneMode',
  'toggleWiFi',
  'toggleData',
  'toggleLocationServices',
  'getGeolocation',
  'setGeolocation',
  'getCurrentDeviceActivity',
  'startDeviceActivity',
  'getAppiumSettings',
  'setAppiumSettings',
  'getCurrentContext',
  'selectContext',
  'getScreenOrientation',
  'setScreenOrientation',
  'isDeviceLocked',
  'lockDevice',
  'unlockDevice',
  'installApp',
  'isAppInstalled',
  'removeApp',
  'pullFileFromDevice',
  'pullFolderFromDevice',
  'pushFileToDevice',
  'listContexts',
  'uploadFile',
  'switchToParentFrame',
  'fullscreen',
  'sendAppToBackground',
  'closeApp',
  'getAppStrings',
  'launchSession',
  'resetApp',
  'hideSoftKeyboard',
  'getDeviceTime',
  'openDeviceNotifications',
  'rotationGesture',
  'shakeDevice' ]
```

`Object.keys(browser)` gives:
```
[ 'controlFlow',
  'schedule',
  'setFileDetector',
  'getExecutor',
  'getSession',
  'getCapabilities',
  'quit',
  'actions',
  'touchActions',
  'executeScript',
  'executeAsyncScript',
  'call',
  'wait',
  'sleep',
  'getWindowHandle',
  'getAllWindowHandles',
  'getPageSource',
  'close',
  'getCurrentUrl',
  'getTitle',
  'findElementInternal_',
  'findElementsInternal_',
  'takeScreenshot',
  'manage',
  'switchTo',
  'driver',
  'element',
  '$',
  '$$',
  'baseUrl',
  'getPageTimeout',
  'params',
  'resetUrl',
  'debugHelper',
  'ready',
  'trackOutstandingTimeouts_',
  'mockModules_',
  'ExpectedConditions',
  'plugins_',
  'allScriptsTimeout',
  'getProcessedConfig',
  'forkNewDriverInstance',
  'restart',
  'restartSync',
  'internalRootEl',
  'internalIgnoreSynchronization' ]
```
There doesn't appear to be a  `.url` method on either one.  Fishing around a bit more to see whats up.

Comment (spadgett) [inline test/integration/helpers.js]: @benjaminapetersen Object.keys won't find properties in the prototype chain, which might explain it

Comment (benjaminapetersen) [inline test/integration/helpers.js]: Ah, thats right, own enumerable only.  Welp, guess thats not getting us anywhere.

**Pull request closed without merging**